pkg/avancement/service_manager.go

```diff
@@ -194,15 +215,15 @@ func encode(gitURL, branch string) string {
 	return url.QueryEscape(gitURL) + "-" + url.QueryEscape(branch)
 }
 
-func addCredentialsIfNecessary(s string, a *git.Author) (string, error) {
-	parsed, err := url.Parse(s)
+func addCredentialsIfNecessary(repoURL string, author *git.Author) (string, error) {
+	parsed, err := url.Parse(repoURL)
 	if err != nil {
-		return "", fmt.Errorf("failed to parse git repo url %v: %w", s, err)
+		return "", fmt.Errorf("%s", "failed to parse git repository url")
```

Comment:
This could be reduced to `errors.New("failed to parse git repository url")`

Again, it's losing the actual error message, which would likely make debugging hard.

Reply:
Yeah so in this case I figured it might actually contain the full URL as well with the token in, but since we'd get an error trying to parse the URL (so we could safely remove the User part of it), I find myself in this odd situation. I'll check now to make sure that is the case

Reply:
What's the latest on this @a-roberts ? Is `err` unsafe to print at this point?